### PR TITLE
feat: add profile page

### PR DESCRIPTION
**Pull request by jiejasonliu**
## note to self: base off main before merging

this PR expands on the profile features and adds the profile page
it doesn't follow the mock to a tee, but if you have any suggestions/changes requested, holler out

refer to the included video as the description of changes

---

https://user-images.githubusercontent.com/29434693/192081029-93e44c49-eaa9-496b-92e9-079cefb5f4ba.mp4

### CODE REVIEW

File: src/assets/_colors.scss [hunk 1]
```diff
@@ -17,4 +18,14 @@ $blue-hover: #1e1f31;
 $light-placeholder: #636b8b;
 $dark-placeholder: #595c80;
 
-$light-red: #e16060;
+$blue-gradient: linear-gradient(
```
Comment (derricklee43) [inline src/assets/_colors.scss:21]: damn! That pulsing gradient is looking dope!

File: src/pages/profile-page/profile-page.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        />
+      );
+    });
+  }
+
+  function _reduceDecksByAccess() {
+    return sortedDecks.reduce(
+      (result: Record<'privateDecks' | 'publicDecks', Deck[]>, deck: Deck) => {
+        const accessLevel = deck.metaData.access;
+        const key = accessLevel === 'Private' ? 'privateDecks' : 'publicDecks';
```
Comment (derricklee43) [inline src/pages/profile-page/profile-page.tsx:93]: Wait, do we already have the access level stuff set up??

**Review by derricklee43 — APPROVED** [review]
This looks really; I like it! The new bubble-divider variant is dope; I'll probably swap it for that in study lesson results

File: src/pages/profile-page/profile-page.tsx (hunk 2 above)
Comment (jiejasonliu) [reply src/pages/profile-page/profile-page.tsx:93]: all decks are Public, but we have no endpoints to alter it atm.

Comment (derricklee43) [reply src/pages/profile-page/profile-page.tsx:93]: oh, haha. I was thinking we were slacking on the front-front editor, lol. But we can just add a dropdown/radio button once they get that endpoint up in the deck editor

Comment (jiejasonliu) [reply src/pages/profile-page/profile-page.tsx:93]: i'm actually completely wrong `POST /Decks/:id` can modify the access level
so yeah, this is something we just need to add to the metadata dropdown

Comment (derricklee43) [reply src/pages/profile-page/profile-page.tsx:93]: Sounds good. I'll make a PR for that

**Pull request merged**